server/src/main/java/org/elasticsearch/bootstrap/BootstrapContext.java

```diff
@@ -20,22 +20,36 @@
 
 import org.elasticsearch.cluster.metadata.MetaData;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
 
 /**
  * Context that is passed to every bootstrap check to make decisions on.
  */
 public class BootstrapContext {
     /**
-     * The nodes settings
+     * The node's environment
      */
-    public final Settings settings;
+    private final Environment environment;
+
     /**
-     * The nodes local state metadata loaded on startup
+     * The node's local state metadata loaded on startup
      */
-    public final MetaData metaData;
+    private final MetaData metaData;
 
-    public BootstrapContext(Settings settings, MetaData metaData) {
-        this.settings = settings;
+    public BootstrapContext(Environment environment, MetaData metaData) {
+        this.environment = environment;
         this.metaData = metaData;
     }
+
+    public Environment getEnvironment() {
```

Comment:
Why do we need getters? These are all final and immutable

Reply:
oh, I thought we would want to avoid that pattern. Do you think 
upgrading `environment.settings()` to be set as an instance variable to BootstrapContext 
is also reasonable?

Reply:
Ah, I see the dilemma. If you want to encapsulate for that reason, that seems fine. Then please use the naming convention proposed in #14266? ie `settings()`, `environment()`, etc since they are read only

Reply:
Alternatively, you could also change all the uses of `context.settings` to `context.environment.settings()`

Reply:
OK, I prefer that proposed approach for immutable method access and will go with that here

Reply:
The reason I will make all of them immutable getter methods of the form `settings()`, `environment()`, `metaData()` is because I do not like `context.environment.settings()` and I do not like that something like `settings()` would be a method, while `metaData` would be a direct access via `context.metaData`. That inconsistent access into these member fields feels weird